hv.c

```diff
@@ -4120,6 +4125,21 @@ Perl_cop_store_label(pTHX_ COP *const cop, const char *label, STRLEN len,
         = refcounted_he_new_pvs(cop->cop_hints_hash, ":", labelsv, 0);
 }
 
+/* share doesn't mean "shared" but "shared string table" */
+/* TODO: should this be atomic increment ??? */
+const struct hek *
+share_hek_hek (const struct hek *hek)
+{
+    ++(
+        ((struct shared_he *)(((char *)hek)
```

Comment:
Why take a `const struct hek *` if you're just going to cast away constness here?

Reply:
irrelevant note: that was earlier implementation of macro

reason for cast away is that this expression access memory before given pointer - hence it doesn't modify context of `struct hek *` but content of `struct share_he`, of which member address is value of given argument